core/src/main/java/bisq/core/dao/monitoring/BlindVoteStateMonitoringService.java

```diff
@@ -257,8 +257,10 @@ private boolean maybeUpdateHashChain(int blockHeight) {
 
         periodService.getCycle(blockHeight).ifPresent(cycle -> {
             List<BlindVote> blindVotes = blindVoteListService.getConfirmedBlindVotes().stream()
-                    .filter(e -> periodService.isTxInCorrectCycle(e.getTxId(), blockHeight))
-                    .sorted(Comparator.comparing(BlindVote::getTxId)).collect(Collectors.toList());
+                    .filter(e -> periodService.isTxInPhaseAndCycle(e.getTxId(), DaoPhase.Phase.BLIND_VOTE, blockHeight))
+                    .filter(e -> e.getTxId() != null)
```

Comment:
I think it would be better to reverse the order of the filters to filter out null txid before using it.

Otherwise it looks correct.

Reply:
Ah good point.